framework/src/utils/SlepcSupport.C

```diff
@@ -127,115 +127,127 @@ getSlepcEigenProblemValidParams()
+    Moose::PetscSupport::setSinglePetscOption(solver_params._prefix + "ksp_max_it",
                                               stringify(params.get<unsigned int>("l_max_its")));
 
-    Moose::PetscSupport::setSinglePetscOption("-ksp_rtol", stringify(params.get<Real>("l_tol")));
+    Moose::PetscSupport::setSinglePetscOption(solver_params._prefix + "ksp_rtol",
+                                              stringify(params.get<Real>("l_tol")));
 
-    Moose::PetscSupport::setSinglePetscOption("-ksp_atol",
+    Moose::PetscSupport::setSinglePetscOption(solver_params._prefix + "ksp_atol",
                                               stringify(params.get<Real>("l_abs_tol")));
   }
   else
   { // linear eigenvalue problem
     // linear solver
-    Moose::PetscSupport::setSinglePetscOption("-st_ksp_max_it",
+    Moose::PetscSupport::setSinglePetscOption(solver_params._prefix + "st_ksp_max_it",
                                               stringify(params.get<unsigned int>("l_max_its")));
 
-    Moose::PetscSupport::setSinglePetscOption("-st_ksp_rtol", stringify(params.get<Real>("l_tol")));
+    Moose::PetscSupport::setSinglePetscOption(solver_params._prefix + "st_ksp_rtol",
+                                              stringify(params.get<Real>("l_tol")));
 
-    Moose::PetscSupport::setSinglePetscOption("-st_ksp_atol",
+    Moose::PetscSupport::setSinglePetscOption(solver_params._prefix + "st_ksp_atol",
                                               stringify(params.get<Real>("l_abs_tol")));
   }
 }
 
 void
 setEigenProblemSolverParams(EigenProblem & eigen_problem, const InputParameters & params)
 {
-  const std::string & eigen_problem_type = params.get<MooseEnum>("eigen_problem_type");
-  if (!eigen_problem_type.empty())
-    eigen_problem.solverParams()._eigen_problem_type =
-        Moose::stringToEnum<Moose::EigenProblemType>(eigen_problem_type);
-  else
-    mooseError("Have to specify a valid eigen problem type");
-
-  const std::string & which_eigen_pairs = params.get<MooseEnum>("which_eigen_pairs");
-  if (!which_eigen_pairs.empty())
-    eigen_problem.solverParams()._which_eigen_pairs =
-        Moose::stringToEnum<Moose::WhichEigenPairs>(which_eigen_pairs);
-
-  // Set necessary parametrs used in EigenSystem::solve(),
-  // i.e. the number of requested eigenpairs nev and the number
-  // of basis vectors ncv used in the solution algorithm. Note that
-  // ncv >= nev must hold and ncv >= 2*nev is recommended
-  unsigned int n_eigen_pairs = params.get<unsigned int>("n_eigen_pairs");
-  unsigned int n_basis_vectors = params.get<unsigned int>("n_basis_vectors");
-
-  eigen_problem.setNEigenPairsRequired(n_eigen_pairs);
-
-  eigen_problem.es().parameters.set<unsigned int>("eigenpairs") = n_eigen_pairs;
-
-  // If the subspace dimension is too small, we increase it automatically
-  if (subspace_factor * n_eigen_pairs > n_basis_vectors)
+  for (const auto i : make_range(eigen_problem.numNonlinearSystems()))
```

Comment:
this is fine for now, however our average eigen problem is 1 eigensystem (neutronics) and a bunch of regulart nonlinear systems (CFD, heat conduction). Our current eigenproblem is probably not coded to solve that  atm